python/ray/data/datasource/torch_datasource.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import TYPE_CHECKING, Callable, List
+
+from ray.data.block import Block, BlockMetadata, T
+from ray.data.datasource import Datasource, ReadTask
+
+if TYPE_CHECKING:
+    import torch.utils.data
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleTorchDatasource(Datasource[T]):
+    """A datasource that let's you use Torch datasets with Ray Data.
+
+    .. warning::
+        ``SimpleTorchDatasource`` doesn't support parallel reads. You should only use
+        this datasource for small datasets like MNIST or CIFAR.
+
+    Example:
+        >>> dataset_factory = lambda: torchvision.datasets.MNIST("data", download=True)
+        >>> dataset = ray.data.read_datasource(
+                SimpleTorchDatasource(), parallelism=1, dataset_factory=dataset_factory
+            )
+        >>> next(dataset.iter_batches(batch_size=2))
+        [(<PIL.Image.Image image mode=L size=28x28 at 0x1142CCA60>, 5)]
+    """
+
+    def prepare_read(
+        self, parallelism: int, dataset_factory: Callable[[], torch.utils.data.Dataset]
```

Comment:
Specify the type as a string so it's not a hard dependency for all Datasets users.
```suggestion
        self, parallelism: int, dataset_factory: Callable[[], "torch.utils.data.Dataset"]
```

Reply:
Fixed